### PR TITLE
30easy Tubka

### PR DESCRIPTION
**Pull request by Tubka**

### CODE REVIEW

File: easy/2.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
-function sayHelloBye (name, num) {
-	
+function sayHelloBye(name, num) {
```
Comment (hayuna) [inline easy/2.js:1]: You can use ternary operator

File: easy/4.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
 function matchHouses(step) {
-	
+  if (step < 1) return step * 6;
```
Comment (hayuna) [inline easy/4.js]: step * 6 for 0 in argument? Why not just 0?

File: easy/6.js [hunk 3]
```diff
@@ -1,10 +1,980 @@
 function amplify(num) {
-	
+  let amplifyNum = [];
```
Comment (hayuna) [inline easy/6.js]: You can use map

File: easy/8.js [hunk 4]
```diff
@@ -1,9 +1,14 @@
 function societyName(friends) {
-	
+  let name = "";
```
Comment (hayuna) [inline easy/8.js]: You can use map

File: easy/9.js [hunk 5]
```diff
@@ -1,13 +1,18 @@
 function hashPlusCount(str) {
-	
+  let hash = (plus = 0);
```
Comment (hayuna) [inline easy/9.js]: You can use regex

File: easy/12.js [hunk 6]
```diff
@@ -1,7 +1,19 @@
 function convertToDecimal(perc) {
-	
+  let decimal = [];
```
Comment (hayuna) [inline easy/12.js]: You can use map

File: easy/14.js [hunk 7]
```diff
@@ -1,9 +1,15 @@
 function convertBinary(str) {
-	
+  let binaryNum = "";
```
Comment (hayuna) [inline easy/14.js]: You can use regex instead of charcodes

File: easy/15.js [hunk 8]
```diff
@@ -1,9 +1,13 @@
 function isOmnipresent(arr, val) {
-	
+  let num = 0;
```
Comment (hayuna) [inline easy/15.js]: You can use every

File: easy/17.js [hunk 9]
```diff
@@ -1,10 +1,10 @@
 function firstVowel(str) {
-	
+  return str.indexOf(str.match(/[aeiou]/gi)[0]);
```
Comment (hayuna) [inline easy/17.js]: Uncaught TypeError: Cannot read property '0' of null - error if there is nl vowel. You cannot use index when you are not sure that array/string has any value. Better remove flag global (g) and index. Then it should return first appearance of vowel

File: easy/18.js [hunk 10]
```diff
@@ -1,8 +1,13 @@
 function reverseCase(str) {
-	
+  let reverse = "";
```
Comment (hayuna) [inline easy/18.js]: You can use map

File: easy/20.js [hunk 11]
```diff
@@ -1,13 +1,15 @@
 function sumOfCubes(nums) {
-	
+  let tri = 0;
```
Comment (hayuna) [inline easy/20.js]: You can use map

File: easy/23.js [hunk 12]
```diff
@@ -1,8 +1,10 @@
 function factorial(n) {
-	
+  let strong = 1;
```
Comment (hayuna) [inline easy/23.js]: You can use reduce

File: easy/24.js [hunk 13]
```diff
@@ -1,14 +1,21 @@
 function filterArray(arr) {
-	
+  let deleteString = _.remove(arr, function(x) {
```
Comment (hayuna) [inline easy/24.js]: Plus for using lodash, you can remove this variable deleteString, use arrow function and shorthand without "return" and curly brackets

File: easy/25.js [hunk 14]
```diff
@@ -1,12 +1,12 @@
-function isZipCode(zipcode){
+function isZipCode(zipcode) {
```
Comment (hayuna) [inline easy/25.js:1]: Could you explain using this questionmark?

File: easy/26.js [hunk 15]
```diff
@@ -1,11 +1,17 @@
 function isSymmetrical(num) {
-	
+  let table = [...num.toString()];
```
Comment (hayuna) [inline easy/26.js]: You can convert to array, reverse it and compare new one with original argument

File: easy/27.js [hunk 16]
```diff
@@ -1,19 +1,19 @@
 function myPi(n) {
-	
+  return Math.PI.toFixed(n);
```
Comment (hayuna) [inline easy/27.js]: toFixed return string type, but test method expect float

File: easy/28.js [hunk 17]
```diff
@@ -1,10 +1,14 @@
 function isAvgWhole(arr) {
-	
+  let sum = 0;
```
Comment (hayuna) [inline easy/28.js]: You can use reduce to sum elements

File: easy/29.js [hunk 18]
```diff
@@ -1,7 +1,13 @@
 function transformUpvotes(str) {
-	
+  str = str.split(" ");
```
Comment (hayuna) [inline easy/29.js]: You can use map

File: easy/30.js [hunk 19]
```diff
@@ -1,10 +1,16 @@
-function vreplace(vowel) {
-	
+function vreplace(vowel, char) {
```
Comment (hayuna) [inline easy/30.js:1]: You can use map or replace with regex

**Review by hayuna — CHANGES_REQUESTED** [review]
Please check my comments

File: easy/25.js (hunk 14 above)
Comment (Tubka) [reply easy/25.js:1]: i can't, i'm beginner with regex and don't understand fully this. I used code which I found on the internet. After tests, yes without questionmark also working.

**Comment by hayuna** [comment]
- 4 You can use ternary operator
- 6 Don't declare new variable if you use map. Map return itself modified array. 
- 8 Could you solve this issue without declare new variables?
- 9 You can use forEach instead of basic for
- 12 Please, remove these comments. You don't have to assign new array to variable. Variable parc is redundant
- 14 Please use map here. It will be much better. To regex pattern you can add flag `i` to avoid duplicate uppercase and lowercase letters
- 18 Could you provide me case, when else will be execute? If yes, I will accept it
- 20 Could you use reduce here?
- 23   return _.range(1, n + 1).reduce((x, y) => {
    return x * y;
  });
  You can use shorthand
  return _.range(1, n + 1).reduce((x, y) => x * y);
- 28 You can use reduce to sum elements
- 29 return (e = parseFloat(e) * 1000);

return parseFloat(e) * 1000;

It should be enough. `else` is redundant. You can left only return parseFloat(e). `str = str.map` - you can just return result of str.map

Result: 81.67%

**Comment by hayuna** [comment]
- 8 Could you solve this issue without declare new variables?
- 20 You dont have to assign new output to accumulator. It works well as default. `((x, y) => x + y ** 3, 0)`

Result: 95%

**Comment by hayuna** [comment]
Result: 100%

**Pull request closed without merging**